scripts/gen_release_notes/tests/test_git_parser.py

```diff
@@ -0,0 +1,203 @@
+        ],
+    },
+]
+
+
+@pytest.mark.parametrize("user_input_changed_files", sample_changed_files)
+def test_get_all_changed_files(user_input_changed_files):
+    """
+    Test for verifying that the changed file function returns something instead of None
+    """
+
+    with patch("requests.get"):
+        result = git_parser.get_all_changed_files(
+            user_input_changed_files["repo_owner"],
+            user_input_changed_files["repo_name"],
+            user_input_changed_files["pr_number"],
+        )
+    assert result is not None
+
+
+@pytest.mark.parametrize("user_input_good_remote_urls", sample_good_remote_urls)
+def test_get_repo_information_good(user_input_good_remote_urls):
+    """
+    Test for verifying that the needed repo information returned is valid and has the expected results
+    """
+
+    mocked_repo = Mock()
+    mocked_repo_url = Mock()
+
+    mocked_repo.remotes = {"origin": mocked_repo_url}
+    mocked_repo_url.url = user_input_good_remote_urls["url"]
+    result = git_parser.get_repo_information(mocked_repo)
+
+    assert result is not None
+    assert (
+        result["owner_name"] == user_input_good_remote_urls["owner"]
+        and result["repo_name"] == user_input_good_remote_urls["repo"]
+    )
+
+
+@pytest.mark.parametrize("user_input_bad_remote_urls", sample_bad_remote_urls)
+def test_get_repo_information_bad(user_input_bad_remote_urls):
+    """
+    Test to check when a failed attempt comes through for getting repo information
+    """
+
+    mocked_repo = Mock()
+    mocked_repo_url = Mock()
+
+    mocked_repo.remotes = {"origin": mocked_repo_url}
+    mocked_repo_url.url = user_input_bad_remote_urls["url"]
+    result = git_parser.get_repo_information(mocked_repo)
+
+    assert result is not None
+    assert (
+        result["owner_name"] != user_input_bad_remote_urls["owner"]
+        or result["repo_name"] != user_input_bad_remote_urls["repo"]
+    )
+
+
+@pytest.mark.parametrize("user_input_pr_information", sample_pr_information)
+def test_get_pr_information(user_input_pr_information):
+    """
+    Test to check if the information gathered in a PR is valid and the data structures are present
+    """
+
+    result = git_parser.get_pr_information(user_input_pr_information)
+
+    assert result is not None
+    assert "pr_number" in result and "pr_title" in result
+
+
+@pytest.mark.parametrize("user_input_pr_type", sample_pr_type)
+def test_get_pr_type(user_input_pr_type):
+    """
+    Test to confirm that the PR type that is gotten from the function meets the standard
+    """
+
+    with patch.object(git_parser, "get_all_changed_files", return_value=[{"filename": "somefile"}]):
+        result = git_parser.get_pr_change_type(
+            user_input_pr_type["repo_owner"],
+            user_input_pr_type["repo_name"],
+            user_input_pr_type["pr_number"],
+            user_input_pr_type["pr_commit_message"],
+        )
+
+        assert result is not None
```

Comment:
I think this none assert is redundant with your next assert.

Reply:
Hm, I see what you mean here, originally this assert is to verify that one of the potential outputs of the function is NOT `None`, since that can be one of the potential outputs for this script. However, the following assert implies that results is already not `None`, so I can clean this out.